back/routes/visitor.js

```diff
@@ -1,74 +1,63 @@
 require('dotenv/config');
 var express = require('express');
-const { Parse } = require('./../parse');
+const {
+  UserDataQuery,
+  LoginQuery,
+  GeneralSignUpQuery,
+  InitializePreferencesQuery,
+} = require('../queries/visitor');
 var router = express.Router();
 
-// Log In
+// -_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_
+// POST the user has loggedIn into Session Table
+// Used in Login.jsx to check information on db and return which kind of user is logging in (adventurer or experience maker)
 router.post('/logIn', async (req, res) => {
-  if (req.body.username === '' || req.body.password === '') {
-    res.status(400);
-    res.send({ error: { message: 'Fill all fields' } });
+  if (!req.headers.username || !req.headers.password) {
+    res.status(400).send({ error: { message: 'Fill all fields' } });
   } else {
-    try {
-      const user = await Parse.User.logIn(req.body.username, req.body.password);
-      res.send({ user });
-    } catch (error) {
-      res.status(400);
-      res.send({ error });
-    }
+    const user = await LoginQuery(req.headers.username, req.headers.password);
+    res.status(200).send({ user });
   }
 });
 
-// Sign Up
+// -_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_
+// POST a new entry for the user table and
+//    if adventurer, also post default values in UserPreference table
+// Used in SignUp.jsx
 router.post('/signUp', async (req, res) => {
-  const hasAllFields = Object.keys(req.body).reduce(function (
-    hasAllFields,
-    inputKey,
-  ) {
-    if (req.body[inputKey] === '') return false;
-    return hasAllFields;
-  },
-  true);
+  // Check that all input fields != undefined/null/empty
+  const hasAllFields = Object.keys(req.body).every((inputKey) => {
+    return (req.body[inputKey]);
+  });
```

Comment:
nit: Validations, although the client might send the right values right now, you'll typically want to validate the specific arguments rather than assuming the req.body contains the correct keys. Again, if you don't have time not a priority, but this comment might be relevant here as well https://github.com/carlosvegap/capstone-kickoff/pull/4#discussion_r918369797

Reply:
Thanks a lot for that point, I will create an issue for this and will try to address it asap!